### PR TITLE
K8S-RELEASE: Adding EKS, GKS, AKS

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

This is a first pass on having webdocs for deploying the MinIO Operator + Tenants on the major cloud kubernetes providers.

The goals are simple:

- To have a dedicated docs URL for collecting provider-specific guidance on deploying the Operator + Tenants
- To have a space for creating dedicated tutorials (where necessary) to account for divergent behavior between K8s flavors on the public clouds
- To reduce support burden by offloading common ops/questions on these K8s distros to our docs
- To have SEO for those search queries

# Scope

For this first pass, the main work is:

- To have installation instructions for the Operator
- To call out basic requirements around CSIs and machine types for each provider
- To call out the Marketplace offering and link to it

# Out of Scope

- Addressing all other tutorials and modifying them to be distro-specific
- Actually testing in depth, as that requires spinning up a bunch of clusters - scheduled for this year

# Extra Bits

- Added a YAML parser for simplifying the collection of excluded URLs per build

# Staged

## Azure Kubernetes Service

- http://192.241.195.202:9000/staging/K8S-RELEASE/aks/index.html
- http://192.241.195.202:9000/staging/K8S-RELEASE/aks/operations/installation.html
- http://192.241.195.202:9000/staging/K8S-RELEASE/aks/operations/install-deploy-manage/deploy-minio-tenant.html

## Elastic Kubernetes Service

- http://192.241.195.202:9000/staging/K8S-RELEASE/eks/index.html
- http://192.241.195.202:9000/staging/K8S-RELEASE/eks/operations/installation.html
- http://192.241.195.202:9000/staging/K8S-RELEASE/eks/operations/install-deploy-manage/deploy-minio-tenant.html

## Google Kubernetes Engine

- http://192.241.195.202:9000/staging/K8S-RELEASE/gke/index.html
- http://192.241.195.202:9000/staging/K8S-RELEASE/gke/operations/installation.html
- http://192.241.195.202:9000/staging/K8S-RELEASE/gke/operations/install-deploy-manage/deploy-minio-tenant.html

### CODE REVIEW

File: source/includes/aks/deploy-minio-on-azure-kubernetes-service.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Prerequisites
+-------------
+
+Existing AKS Cluster
+~~~~~~~~~~~~~~~~~~~~
+
+This procedure assumes an existing :abbr:`AKS (Azure Kubernetes Service)` cluster onto which you can deploy the MinIO Operator.
+
+The Operator by default deploys pods and services with two replicas each and pod anti-affinity.
+The GKE cluster should therefore have at least two nodes available for scheduling Operator pods and services.
```
Comment (djwfyi) [inline source/includes/aks/deploy-minio-on-azure-kubernetes-service.rst:42]: Swap `GKE` for `AKS`

**Review by djwfyi — CHANGES_REQUESTED** [review]
We'll need to make sure to address Algolia after merging this PR.

A few copy/paste gotchas on different platforms that I noted.
The opening section of the index.rst page is bugging me with each of the first six sentences being separate paragraphs. Suggest reworking that for a better reader experience for our landing page.

Otherwise, good work!
Let me know if you want me to do another pass at the index page. Otherwise, I can LGTM it.

**Comment by pjuarezd** [comment]
So far looks good, I would point out some copy-paste little happy accidents, but @djwfyi already did a great job on that.

I've noticed that is no mention of the billing and Install methods offerings for each cloud provider, but that is somenthing that we can help extend, having a section for the Marketplaces in the docs is great!

**Comment by ravindk89** [comment]
@pjuarezd that's step two - technically we have the install instructions linked, but I'd like to onboard them as well + test them myself to see how exactly it works (e.g. if I connect an EKS cluster via marketplace and spin up two MinIO Tenants, does it pick up both tenants automatically? Things like that)

Pricing probably we will leave in the marketplace, but we can mention in the relevant section that Marketplace billing is based on total amount of storage used with a minimum of 100TB.

**Pull request merged**